src/lib/apparmor/profiles.rb

```diff
@@ -23,13 +25,13 @@ def initialize(name, status)
 
     # Set to complain mode
     def complain
-      system("/usr/sbin/aa-complain #{@name}")
+      execute("/usr/sbin/aa-complain", @name)
       @status = 'complain'
     end
 
     # Set to enforce mode
     def enforce
-      system("/usr/sbin/aa-enforce #{@name}")
+      execute("/usr/sbin/aa-enforce", @name)
```

Comment:
just note, it ignore return value, so if switch failed, it is silently.

Reply:
I have just taken the code as-is from sle-15-sp1.